src/com/google/enterprise/adaptor/database/DatabaseAdaptor.java

```diff
@@ -317,18 +319,21 @@ public void init(AdaptorContext context) throws Exception {
    * @param sql the SQL query; the query is prepared, but not executed
    * @param columnConfig the configuration property for the column names
    * @param columnNames the column names to verify
+   * @return a Map of columnName to Integer java.sql.Types
```

Comment:
can that be Map<String, UniqueKey.ColumnType>?  It does seem a bit unfortunate that SQL types are Integer. What do you think?